src/scripts/Controller.js

```diff
@@ -1196,21 +1197,68 @@ export default class SitemapController {
 			type: 'SelectorText',
 			multiple: false,
 		});
-
 		this._editSelector(selector, sitemap);
 	}
 
+	async showConfirmActionPanel(message) {
+		const $actionConfirm = ich.ActionConfirm();
+		$('#viewport').append($actionConfirm);
+		Object.entries(message).forEach(selector => $(selector[0]).attr('data-i18n', selector[1]));
+		$actionConfirm.modal('show');
+		Translator.translatePage();
+	}
+
+	waitConfirmActionPanel() {
+		let promise = new Promise(resolve => {
+			$('#submit').click(() => resolve(true));
+			$('#cancel,.modal-backdrop').click(event => resolve(false));
+		});
+		return promise.then(value => {
+			$('.modal, .modal-backdrop').remove();
+			$('.modal-open').removeClass('modal-open');
+			return value;
+		});
+	}
+
 	async deleteSelector(button) {
-		const sitemap = this.state.currentSitemap;
 		const selector = $(button).closest('tr').data('selector');
-		sitemap.deleteSelector(selector);
-		await this.store.saveSitemap(sitemap);
+		const sitemap = this.state.currentSitemap;
+		const childCount = sitemap.getDirectChildSelectors(selector.id).length;
+		const replace = {
+			'#child-count': childCount ? childCount : '',
+			'#selector-id': selector.id,
+		};
+		const attributes_i18n = {
+			'.modal-title': 'modal_confirm_action_title_delete_selector',
+			'.modal-subtitle': 'modal_confirm_action_child_delete',
+			'#submit': 'modal_confirm_action_submit',
+			'#cancel': 'modal_confirm_action_cancel',
+		};
+		this.showConfirmActionPanel(attributes_i18n);
+		Object.entries(replace).forEach(selector => $(selector[0]).append(selector[1]));
```

Comment:
just do `$('#selector-id').append(selector.id)` explicitly
same in delete sitemap